src/common/number/format_number.ts

```diff
@@ -90,8 +91,15 @@ export const formatNumber = (
  * @returns An `Intl.NumberFormatOptions` object with `maximumFractionDigits` set to 0, or `undefined`
  */
 export const getNumberFormatOptions = (
-  entityState: HassEntity
+  entityState: HassEntity,
+  entity: EntityRegistryEntry | undefined
```

Comment:
```suggestion
  entityState: HassEntity,
  entity?: EntityRegistryEntry
```
So you can avoid passing `undefined` in tests.